src/log.h

```diff
@@ -221,16 +279,62 @@ class StreamProxy {
 public:
 	StreamProxy(std::ostream *os) : m_os(os) { }
 
+	inline static void maybe_fix_stream_state(std::ostream &os)
+	{
+		std::ios::iostate state = os.rdstate();
+		if (state != std::ios::goodbit) {
+			// clear error state so the stream works again
+			os.clear();
+			if (state & std::ios::eofbit)
+				os << "(ostream:eofbit)";
+			if (state & std::ios::badbit)
+				os << "(ostream:badbit)";
+			if (state & std::ios::failbit)
+				os << "(ostream:failbit)";
+		}
+	}
+
 	template<typename T>
-	StreamProxy& operator<<(T&& arg) {
+	StreamProxy& operator<<(T&& arg)
+	{
 		if (m_os) {
+			maybe_fix_stream_state(*m_os);
```

Comment:
don't you want to do this directly after?

Reply:
If this were the only way to access m_os, there wouldn't be any functional difference between doing it before or after, since each call to operator<< is directly followed by another.

But `LogStream::operator std::ostream&` exposes m_os directly. So it is possible for an error state to occur in between calls to `StreamProxy::operator<<`. Because of this, doing `maybe_fix_stream_state` before has the advantage of fixing the stream before output is lost due to an error state.